src/chocolatey.tests.integration/scenarios/UpgradeScenarios.cs

```diff
@@ -3525,5 +3525,105 @@ public void should_not_have_executed_beforemodify_hook_script_for_upgrade_versio
                 MockLogger.contains_message("pre-beforemodify-all.ps1 hook ran for upgradepackage 1.1.0", LogLevel.Info).ShouldBeFalse();
             }
         }
+
+        public class when_upgrading_a_package_with_beforeModify_script_with_dependencies_with_beforeModify_scripts : ScenariosBase
+        {
+            private const string TargetPackageName = "hasdependencywithbeforemodify";
+            private const string DependencyName = "isdependencywithbeforemodify";
+
+            public override void Context()
+            {
+                base.Context();
+
+                Scenario.add_packages_to_source_location(Configuration, "{0}.*".format_with(TargetPackageName) + Constants.PackageExtension);
+                Scenario.add_packages_to_source_location(Configuration, "{0}.*".format_with(DependencyName) + Constants.PackageExtension);
+                Scenario.install_package(Configuration, DependencyName, "1.0.0");
+                Scenario.install_package(Configuration, TargetPackageName, "1.0.0");

```

Comment:
Would it be an idea to extract these version numbers into a variable of the class in case they change?

(This comment would apply equally to the rest of the version numbers in the tests here)

Reply:
In general, unless the test structure is changing, the version numbers shouldn't need to change. The only reason the package IDs got put in properties is because I had to reference them like 10 times each lol